src/test/java/core/basesyntax/operation/BalanceHandlerTest.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.operation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.ShopServiceImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.transaction.FruitTransaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class BalanceHandlerTest {
+    private static final String APPLE = "apple";
+    private static final int ONE_HUNDRED_QUANTITY = 100;
+    private ShopService shopService;
+    private BalanceHandler balanceHandler;
+    private FruitTransaction fruitTransaction;
+    private OperationStrategy operationStrategy;
+
+    @BeforeEach
+    void setUp() {
+        shopService = new ShopServiceImpl(operationStrategy);
+        balanceHandler = new BalanceHandler(shopService);
+        fruitTransaction = new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                APPLE, ONE_HUNDRED_QUANTITY);
+    }
+
+    @Test
+    void apply_shouldAddFruitsToShopService() {
+        balanceHandler.apply(fruitTransaction);
+        assertEquals(ONE_HUNDRED_QUANTITY, shopService.getQuantity(APPLE));
+    }
+
+    @Test
+    void apply_nonBalanceOperation_throwsRuntimeException() {
+        fruitTransaction = new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                APPLE, ONE_HUNDRED_QUANTITY);
+

```

Comment:
Let’s remove this test after refactoring. The current decision contradicts the implementation of the pattern strategy. Check other test classes